root/static/scripts/edit/URLCleanup.js

```diff
@@ -3976,6 +3976,45 @@ const CLEANUPS: CleanupEntries = {
       return {result: false, target: ERROR_TARGETS.URL};
     },
   },
+  'tidal': {
+    match: [new RegExp(
+      '^(https?://)?' +
+      '(([^/]+\\.)*(desktop|listen|stage|store|www)\\.)?tidal\\.com' +
+      '/.*(album|artist|track|video)/',
+      'i',
+    )],
+    restrict: [LINK_TYPES.streamingpaid],
+    clean: function (url) {
+      url = url.replace(/^(?:https?:\/\/)?(?:(?:[^\/]+\.)*(?:desktop|listen|stage|store|www)\.)?tidal\.com\/(?:#!\/)?([\w\/]+).*$/, 'https://tidal.com/$1');
+      url = url.replace(/^https:\/\/tidal\.com\/(?:[a-z]{2}\/)?(?:browse\/|store\/)?(?:[a-z]+\/\d+\/)?([a-z]+)\/(\d+)(?:\/[\w]*)?$/, 'https://tidal.com/$1/$2');
+      return url;
+    },
+    validate: function (url, id) {
+      const m = /^https:\/\/tidal\.com\/([a-z]+)\/\d+$/.exec(url);
+      if (m) {
+        const prefix = m[1];
+        switch (id) {
+          case LINK_TYPES.streamingpaid.artist:
+            if (prefix === 'artist') {
+              return {result: true};
+            }
+            return {result: false, target: ERROR_TARGETS.ENTITY};
```

Comment:
Is there no use on having an error message you think?

Reply:
I thought there would be a generic one instead, but it's not the case, will have to open a pull request for that too.

Reply:
Created https://tickets.metabrainz.org/browse/MBS-11977 and started working on it.

Reply:
Cherry-picked https://github.com/metabrainz/musicbrainz-server/pull/2263/commits/e4da3d6caa84d9307a1123a8b02b8f49523cde41 from pull request #2280.